src/prime_rl/trainer/config.py

```diff
@@ -40,14 +40,70 @@ class ModelConfig(BaseConfig):
     ] = True
 
 
+class BaseSchedulerConfig(BaseModel):
+    """Base configuration for learning rate schedulers."""
+    
+    warmup_steps: Annotated[int, Field(ge=0, description="Number of warmup steps for the learning rate scheduler.")] = 0
+
+
+class ConstantSchedulerConfig(BaseSchedulerConfig):
+    """Configuration for constant learning rate scheduler."""
+    
+    scheduler: Literal["constant"] = "constant"
+
+
+class LinearSchedulerConfig(BaseSchedulerConfig):
+    """Configuration for linear learning rate scheduler."""
+    
+    scheduler: Literal["linear"] = "linear"
+    decay_steps: Annotated[
+        int | None,
+        Field(
+            ge=1,
+            description="Number of steps to decay the learning rate during the final portion of training. If None, will use remaining steps after warmup.",
+        ),
+    ] = None
```

Comment:
wondering if it could not be easier to do a decay percentage like in torchtitan, but no strong opinion

Reply:
I just think it's neat to have warmup and decay standardized as being based on steps rather than ratio over total steps, IMO it is fine like this

Reply:
sure !